### PR TITLE
Network docs: subnet division, updated diagrams & guidance

### PR DESCRIPTION
**Pull request by akarve**

### CODE REVIEW

File: docs/technical-reference.md [hunk 1]
```diff
@@ -70,14 +104,9 @@ application load balancer
 - 2 private subnets for intra-VPC traffic to and from the Quilt RDS database and
 OpenSearch domain
 (1/8 of the VPC CIDR)
+- 2 user subnets (same subnet as for the load balancer)
```
Comment (dimaryaz) [inline docs/technical-reference.md]: I don't think we create them - we only support them for customer-provided networks. (So I think the TF description got it backwards.) There's also no more space left - we already have 1/2, 1/4, 1/8, and 1/8 unused.

File: docs/technical-reference.md [hunk 2]
```diff
@@ -21,47 +21,81 @@ connect with other users
...
+> For cost-sensitive deployments, Quilt ECS services can be configured to use
+> a single AZ.
+
+> You may use a combination of interface endpoints and gateway endpoints to
+> restrict the data plane traffic shown above to your VPC.
+> See [Private endpoint access](advanced-features/private-endpoint-access.md) for more.
+
+We recommend a CIDR block no smaller than /25 (128 addresses) for production
+deployments. Larger CIDR blocks are easier to upgrade to new Quilt versions
+with expanded services.
```
[the remark this answers is not in the document]
Comment (akarve) [reply docs/technical-reference.md:61]: I think ECS will fail but more importantly we know that, say we need more addresses than they have, it's not really possible to change subnets without a lot of footwork.

File: docs/technical-reference.md [hunk 3]
```diff
@@ -21,47 +21,81 @@ connect with other users
...
+| Private    | a  | Routes to Internet     | ECS, Lambda | 16 |
+| Private    | b  | " | " | 16 |
+| Intra      | a  | Does not route to Internet  | RDS, OpenSearch* | 16 |
+| Intra      | b  | " | " | 16 |
+| User       | a  | Reachable by GUI catalog users | App load balancer, API Gateway Endpoint | 4 |
+| User       | b  | " | " | 4 |
+
+> \* One IP per master node, one IP per data node
+
+> † Excludes the 5 IPs per subnet AWS consumes (network, routing, DNS, reserved, broadcast). Includes room for service expansion.
```
[the remark this answers is not in the document]
Comment (akarve) [reply docs/technical-reference.md:76]: OK.

File: docs/technical-reference.md [hunk 4]
```diff
@@ -21,47 +21,81 @@ connect with other users
...
+
+#### Production, multi-AZ subnet division for private ELBv2 (you provide the network)
```
[the remark this answers is not in the document]
Comment (akarve) [reply docs/technical-reference.md:78]: Sure I'll drop this one.

**Pull request merged**